src/Controls/tests/DeviceTests/Elements/Window/WindowTests.cs

```diff
@@ -121,6 +121,105 @@ await CreateHandlerAndAddToWindow<IWindowHandler>(window, async (handler) =>
 		}
 #endif
 
+
+		[Theory]
+		[ClassData(typeof(WindowPageSwapTestCases))]
+		public async Task NavigatedEventsFireOnEachIncomingAndOutGoingPage(WindowPageSwapTestCase swapOrder)
+		{
+			SetupBuilder();
+
+			var firstRootPage = swapOrder.GetNextPageType();
+			var window = new Window(firstRootPage);
+
+			bool hasNavigatedToFired = false;
+			bool hasNavigatedFromFired = false;
+			bool hasNavigatingFromFired = false;
```

Comment:
Can we make these int and then check we got the correct number of events?

Reply:
Hi @mattleibow  , I’ve updated the test cases to verify that the correct number of events are in place. Please review the changes and let me know if any further modifications are required.